picard_mark_duplicates_2.8.1/picard_mark_duplicates_2.8.1.cwl

```diff
@@ -0,0 +1,166 @@
+      clear the default value. Possible values: {STRICT,LENIENT, SILENT}
+  - id: bam_compression_level
+    type: int?
+    inputBinding:
+      position: 0
+      prefix: COMPRESSION_LEVEL=
+      separate: false
+    doc: >-
+      Compression level for all compressed files created (e.g. BAM and GELI). 
+      Default value:5. This option can be set to 'null' to clear the default
+      value.
+  - default: true
+    id: create_bam_index
+    type: boolean?
+    inputBinding:
+      position: 0
+      prefix: CREATE_INDEX=true
+    doc: >-
+      Whether to create a BAM index when writing a coordinate-sorted BAM file. 
+      Default value:false. This option can be set to 'null' to clear the default
+      value. Possible values:{true, false}
+  - default: true
+    id: assume_sorted
+    type: boolean?
+    inputBinding:
+      position: 0
+      prefix: AS=true
+  - id: duplicate_scoring_strategy
+    type: string?
+    inputBinding:
+      position: 0
+      prefix: DUPLICATE_SCORING_STRATEGY=
+    doc: >-
+      The scoring strategy for choosing the non-duplicate among candidates. 
+      Default value:SUM_OF_BASE_QUALITIES. This option can be set to 'null' to
+      clear the default value.Possible values: {SUM_OF_BASE_QUALITIES,
+      TOTAL_MAPPED_REFERENCE_LENGTH, RANDOM}
+  - id: optical_duplicate_pixel_distance
+    type: int?
+    inputBinding:
+      position: 0
+      prefix: OPTICAL_DUPLICATE_PIXEL_DISTANCE=
+    doc: >-
+      The maximum offset between two duplicate clusters in order to consider
+      them optical duplicates. The default is appropriate for unpatterned
+      versions of the Illumina platform. For the patterned flowcell models, 2500
+      is moreappropriate. For other platforms and models, users should
+      experiment to find what works best.  Default value: 100. This option can
+      be set to 'null' to clear the default value.
+outputs:
+  - id: bam
+    type: File
+    outputBinding:
+      glob: '$(inputs.input.basename.replace(/.bam/, ''_md.bam''))'
+    secondaryFiles:
+      - ^.bai
+      - ^.metrics
+label: picard_mark_duplicates_2.8.1
+arguments:
+  - position: 0
+    prefix: ''
+    valueFrom: "${\n  if(inputs.memory_per_job && inputs.memory_overhead) {\n   \n    if(inputs.memory_per_job % 1000 == 0) {\n    \t\n      return \"-Xmx\" + (inputs.memory_per_job/1000).toString() + \"G\"\n    }\n    else {\n      \n      return \"-Xmx\" + Math.floor((inputs.memory_per_job/1000)).toString() + \"G\" \n    }\n  }\n  else if (inputs.memory_per_job && !inputs.memory_overhead){\n    \n    if(inputs.memory_per_job % 1000 == 0) {\n    \t\n      return \"-Xmx\" + (inputs.memory_per_job/1000).toString() + \"G\"\n    }\n    else {\n      \n      return \"-Xmx\" + Math.floor((inputs.memory_per_job/1000)).toString() + \"G\" \n    }\n  }\n  else if(!inputs.memory_per_job && inputs.memory_overhead){\n    \n    return \"-Xmx15G\"\n  }\n  else {\n    \n  \treturn \"-Xmx15G\"\n  }\n}"
```

Comment:
this is a bit hard to review in a PR but maybe eventually we can think of a way to get this to print more nicely in the CWL

Reply:
![bitmoji](https://render.bitstrips.com/v2/cpanel/e0cbfa2f-8293-4e17-ae61-0460d9c3ff47-091819f5-3b43-4653-8093-982c9e72a960-v1.png?transparent=1&palette=1&width=246)